gradio/components.py

```diff
@@ -419,7 +419,10 @@ def __init__(
 
         #
         self.lines = lines
-        self.max_lines = max_lines if type == "text" else 1
+        if type == "text":
+            self.max_lines = lines if lines > max_lines else max_lines
```

Comment:
If you could, you could do:
```suggestion
            self.max_lines = max(lines, max_lines)
```